apps/web/components/posts/table.tsx

```diff
@@ -1,29 +1,44 @@
 import { useRouter } from "next/router";
-import { Post } from "../../interface/post";
+import { Post, DeletePostNew } from "../../interface/post";
+import { DELETE_POST, GET_POSTS } from "web/components/queries"
+import { useMutation } from "@apollo/client";
 
-interface TableProps {
+
+interface TableProps { //PROPS
   posts: Post[];
 }
 
 const Table = ({ posts }: TableProps) => {
-  const { push } = useRouter();
+  const { push } = useRouter(); //BRINGS THE WEBSITE TO ANOTHER PAGE
+
+  // wrapper function that calls on the DeletePost Function
+  const DeleteFunc = (id: string) => {
+      deletePost({variables: {id: id}})
+      }
+
+  // This is the DeletePost function using the useMutation from the resolvers
+  const [deletePost] = useMutation<DeletePostNew>(DELETE_POST, {refetchQueries: [GET_POSTS]});
```

Comment:
[nit] We could just use the `Post` interface as the return type here. Also, if we also create a `DeletePostVariables` interface for the mutation variables and use them like this
```suggestion
  const [deletePost] = useMutation<Post, DeletePostVariables>(DELETE_POST, {refetchQueries: [GET_POSTS]});
```
then we can get type hints when we call `deletePost({ variables: { id: id }})`. Otherwise, we have to remember what the query variable names should be.